.github/workflows/deploy_to_staging.yml

```diff
@@ -1,41 +1,111 @@
 name: Deploy to Staging
 
 on:
-  push:
-    branches:
-      - master
-    paths-ignore:
-      - 'documentation/**'
-
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Which version are you creating?
+        required: true
+      ref:
+        description: Git ref to deploy
+        required: true
+        default: develop
 jobs:
-  deploy:
+  deploy-to-staging:
     runs-on: ubuntu-20.04
     steps:
-      - name: setup-inputs
+      - uses: actions/checkout@v2
+        name: Checkout Code
+        with:
+          ref: ${{ github.event.inputs.ref }}
+
+      - name: Branch protection OFF
+        uses: octokit/request-action@v2.x
+        with:
+          route: PUT /repos/DFE-Digital/early-careers-framework/branches/master/protection
+          repository: ${{ github.repository }}
+          owner: DFE-Digital
+          required_status_checks: |
+            null
+          required_pull_request_reviews: |
+            null
+          enforce_admins: |
+            null
+          allow_force_pushes: |
+            true
+          required_linear_history: |
+            true
+          allow_deletions: |
+            null
+          restrictions: |
+            null
+        env:
+          GITHUB_TOKEN: ${{ secrets.PERSONAL_TOKEN }}
```

Comment:
Rather than doing it in actions, can we set some protections on `master` branch of our repo that admins don't care about, and use an admin token to bypass them in this action?

Reply:
(I really don't like the idea of unprotecting a branch so we can make it look prettier on a release :P)

Reply:
No, because we're all admins. This is the accepted way of doing it

Reply:
I spent some time looking, and found no combination of branch protection rules that do what we want

Reply:
Someone has even written a bot for it (which I'm not a big fan of) https://github.com/benjefferies/branch-protection-bot

Reply:
So what we're doing here is disabling the extra protections on master - the ones you're advocating for disabling permanently. This branch will still only be writable by people with write permissions on the repo

Reply:
It becomes the same as any other branch for a short time

Reply:
I can't comment on the rest of the PR but this specifically seems like a sensible approach to me

Reply:
Oh, so `null` in the workflow variables means 'do not change this'? I guess that makes sense then.

Reply:
See https://docs.github.com/en/rest/reference/repos#update-branch-protection

Reply:
Discussed offline. This is changing branch protection rules, even when they're all off this branch is as "protected" as any other, only team members have push access